templates/terraform/objectlib/base.go.erb

```diff
@@ -0,0 +1,44 @@
+<%= lines(autogen_notice :go) -%>
+
+package google
+
+<%
+    resource_name = product_ns + object.name
+    properties = object.all_user_properties
+%>
+
+<%# constants usually contain functions called from inside the setup file. -%>
+<%= lines(compile(object.custom_code.constants)) if object.custom_code.constants -%>
+
+func Get<%= resource_name -%>ApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
+    obj := make(map[string]interface{})
+<%  object.settable_properties.each do |prop| -%>
+    <%= prop.api_name -%>Prop, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= prop.name.underscore -%>"), d, config)
```

Comment:
Thinking longer term, I wonder if we want to abstract some templating out of the Terraform provider to a shared location so this also picks up changes like https://github.com/GoogleCloudPlatform/magic-modules/pull/1418

In addition, I think we should clearly define ownership of this MM provider and the downstream repo; going forward, what does the Terraform team own? What does PSO own?